awx/main/models/schedules.py

```diff
@@ -35,6 +35,45 @@
 UTC_TIMEZONES = {x: tzutc() for x in dateutil.parser.parserinfo().UTCZONE}
 
 
+SECONDS_IN_WEEK = 7 * 24 * 60 * 60
+
+
+def fast_forward_date(rrule):
+    '''
+    Utility to fast forward an rrule, maintaining consistency in the resulting
+    occurrences
+    Fast forwards the rrule to 7 days ago
+    Returns a datetime object
+    '''
+    if not rrule._freq in (dateutil.rrule.HOURLY, dateutil.rrule.MINUTELY):
```

Comment:
`not x in y` is a common mistake. Linters would usually require it to be an `x not in y` instead, as it's clearer what the semantics of the check is. I wouldn't even be sure whether it's `(not x) in y` or `not (x in y)` off the top of my head. Let's improve this a bit:
```suggestion
    if rrule._freq not in {dateutil.rrule.HOURLY, dateutil.rrule.MINUTELY}:
```